Makefile

```diff
@@ -96,15 +96,23 @@ docker-push: ## Build a multi-architecture image and push that into the registry
 	$(DOCKER_BUILDX) stop image-builder
 	rm Dockerfile.cross
 
-container-build: module-check
+container-npm-ci:
+	mkdir -p node_modules
+	$(CONTAINER_RUN) --read-only \
+		--mount type=bind,source=$(CURDIR)/node_modules,target=/src/node_modules \
+		--mount type=tmpfs,destination=/var/hugo/.npm,tmpfs-mode=01777 \
+		--mount type=tmpfs,destination=/var/hugo/.cache,tmpfs-mode=01777 \
+		$(CONTAINER_IMAGE) npm ci
```

Comment:
I'd really prefer to bake the NPM modules into the container image.

(the approach in this PR it also doesn't work for me locally using either Docker or Podman; I'm sure I could get it to work, but baking in a correct `/src/node_modules` would be easier for people)

Eventually we'd like to provide a PowerShell alternative for the `Makefile`. If we can keep the complexity inside the container, that's so much better.

I'll see if I can find a contributor willing to work on reshaping this part for you @chalin.

Reply:
> I'd really prefer to bake the NPM modules into the container image.

Ok. Done, PTAL.